### PR TITLE
Moves `node-template` into substrate repo

### PR DESCRIPTION
**Pull request by bkchr**
This pr moves the `node-template` into the substrate repo, so we are informed about stuff that breaks the template.
This pr also provides a tool for creating a release of `node-template`. Release includes adapting the `Cargo.toml` files to point to the current git commit, building the project to ensure everything works and creating a `.tar.gz` file containing the source code.

### CODE REVIEW

File: node-template/build.rs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+// Copyright 2015-2018 Parity Technologies (UK) Ltd.
+// This file is part of Substrate.
```
Comment (gnunicorn) [inline node-template/build.rs]: This license header conflicts with the LICENSE in this directory (two files above), doesn't it?

**Comment by gavofyork** [comment]
need resolving, otherwise happy for this to be merged once:
- CI integration is done
- the getsubstrate.io script is updated to get s-n-t from the upload destination
- paritytech/substrate-node-template repo is deleted

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -93,10 +93,11 @@ build:rust:linux:release:          &build
     - mkdir -p ./artifacts
     - mv ./target/release/substrate ./artifacts/.
     - echo -n "Substrate version = "
-    - ./artifacts/substrate --version | 
+    - ./artifacts/substrate --version |
       sed -n -r 's/^substrate ([0-9.]+.*-[0-9a-f]{7,13})-.*$/\1/p' |
       tee ./artifacts/VERSION
     - sha256sum ./artifacts/substrate | tee ./artifacts/substrate.sha256
+    - ./scripts/node-template-release.sh ./artifacts/substrate-node-template.tar.gz
```
Comment (bkchr) [inline .gitlab-ci.yml]: @gabreal could you verify that this is correct? Can I test this somehow?

**Comment by gavofyork** [comment]
@bkchr feel free to merge once the tooling is all set up.

File: .gitlab-ci.yml (hunk 2 above)
Comment (gabreal) [reply .gitlab-ci.yml]: yes, you can do this like that. the files in artifacts will be released to S3 and also be included in the dockerfile.

Comment (gabreal) [reply .gitlab-ci.yml]: You can test this when you rewrite this file to build it also for pull requests. The publishing although cannot be tested so easily.

**Comment by gavofyork** [comment]
guess this is ready to go in now?

**Comment by bkchr** [comment]
@gabreal is the download url ready?

**Comment by attente** [comment]
Hi, has this change been properly tested against the examples at https://substrate.readme.io/docs/creating-a-custom-substrate-chain and https://substrate.readme.io/docs/substrate-runtime-recipes? Ignoring the re-assignment of `srml_support` to `support`, the node template can't find the `decl_module` or `decl_storage` support macros during compilation:

```
error: cannot find macro `decl_module!` in this scope
  --> /home/william/tmp/recipe/runtime/src/runtime_example.rs:18:1
   |
18 | decl_module! {
   | ^^^^^^^^^^^

error: cannot find macro `decl_storage!` in this scope
  --> /home/william/tmp/recipe/runtime/src/runtime_example.rs:27:1
   |
27 | decl_storage! {
   | ^^^^^^^^^^^^
```

**Comment by bkchr** [comment]
Could you please give more context? It looks like you are not including the macro.
The template is now always checked to compile against master.

**Comment by attente** [comment]
Trying to declare a new runtime module the same way as the tutorials doesn't work any more because the macros aren't in scope. Is there a way to explicitly import them?

**Comment by bkchr** [comment]
@attente `use srml_support::decl_module`. Do you heard of rust 2018? We switched the node template to this.

**Comment by shawntabrizi** [comment]
@attente @bkchr looking at this now

**Comment by shawntabrizi** [comment]
@attente We found some bugs with the node-template, thanks for reporting.

https://github.com/paritytech/substrate/pull/1742 should solve this and it will take a little bit of time to get the changes built into a new file.

I will need to update the tutorial as well to address this

**Pull request merged**